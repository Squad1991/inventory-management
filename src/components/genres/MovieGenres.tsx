import React from 'react';
import Inline from '~src/components/common/primitive/Inline';
import { Genre } from '~src/routes/movie/types';

interface MovieGenresProps {
  genres: Genre[];
}

const MovieGenres: React.FC<MovieGenresProps> = ({ genres }) => {
  return (
    <Inline>
      {genres
        .map(genre => {
          return genre.name;
        })
        .join('/')}
    </Inline>
  );
};

export default MovieGenres;
