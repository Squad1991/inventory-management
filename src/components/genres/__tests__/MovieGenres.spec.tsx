import { render } from '@testing-library/react';
import React from 'react';
import MovieGenres from '~src/components/genres/MovieGenres';

describe('MovieGenres', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <MovieGenres
        genres={[
          {
            id: 1,
            name: 'Action',
          },
          {
            id: 2,
            name: 'Comedy',
          },
        ]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
