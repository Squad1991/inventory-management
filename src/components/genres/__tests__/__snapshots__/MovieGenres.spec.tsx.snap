// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieGenres > should match the snapshot 1`] = `
<div>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}._2lx2vrvc{flex-direction:row}
  </style>
  <div
    class="_1e0c1txw _2lx2vrvc"
  >
    Action/Comedy
  </div>
</div>
`;
